src/api/cvc4cpp.h

```diff
@@ -2559,6 +2559,26 @@ class CVC4_PUBLIC Solver
   Term mkTuple(const std::vector<Sort>& sorts,
                const std::vector<Term>& terms) const;
 
+  /**
+   * returns a term that represents ((_ tupSel index) tuple)
+   * @param index represents the index of the selected element in the tuple
+   * @param tuple a node that has a tuple type
+   * @pre sort of tuple is (Tuple T1 ... Tn) and index < n
+   * @return the element at the given index in tuple
+   */
+  Term tupleSelect(uint32_t index, Term tuple) const;
+
+  /**
+   * returns a tuple that represents
+   * (mkTuple ((_ tupSel i_1) t) ... ((_ tupSel i_n) t))
+   * where each i_j is less than the length of t
+   * @param t a node that has a tuple type
+   * @param indices of the projected elements
+   * @return a tuple that represents the elements with indices {i_1, ..., i_n}
+   * in tuple t
+   */
+  Term tupleProject(Term t, std::vector<uint32_t> indices) const;
```

Comment:
```suggestion
  Term tupleProject(const std::vector<uint32_t>& indices, Term t) const;
```

Reply:
I think t should come after indices to be consistent with tupleSelect.

Reply:
Done. More importantly, do we need this function? If we are introducing the project operator, the user would have 2 options to project a tuple